great_expectations/rule_based_profiler/parameter_builder/parameter_builder.py

```diff
@@ -564,19 +542,21 @@ def get_metrics(
                 parameters=parameters,
             )
 
-        # Step-10: Build and return result to receiver (apply simplifications to cases of single "metric_value_kwargs").
+        # Step-8: Build and return result to receiver (apply simplifications to cases of single "metric_value_kwargs").
+
+        details: dict = {
+            "metric_configuration": {
+                "metric_name": metric_name,
+                "domain_kwargs": domain_kwargs,
+                "metric_value_kwargs": metric_value_kwargs[0]
```

Comment:
Why unwrap the dictionary here if it has 1 value? Seems to break the symmetry in whoever is consuming this. Also what is the type of `metric_value_kwargs`. Since it's named `*_kwargs` I would expect a `dict` but you are accessing with `0`, which can't be a keyword name, it looks like a list?

Reply:
Oh, I see now you've just moved this. Looks odd to me though.

Reply:
@billdirks This compactness is for the future of Batch-Metric Store.